include/seastar/core/shared_future.hh

```diff
@@ -165,6 +169,11 @@ private:
                     _peers.pop_front();
                 }
             }
+            _keepaliver.release();
+        }
+
+        task* waiting_task() noexcept override {
+            return nullptr;
```

Comment:
This is inaccurate, is it not? All of _peers are waiting.

But we don't have an interface to expose multiple waiters. Still, we could have exposed one.